staging/src/k8s.io/client-go/tools/cache/fifo.go

```diff
@@ -215,6 +236,10 @@ func (f *FIFO) Delete(obj interface{}) error {
 	f.lock.Lock()
 	defer f.lock.Unlock()
 	f.populated = true
+	// delete key in `queue`
+	order := f.items[id].(OrderItem).Order
+	f.queue = append(f.queue[:order], f.queue[order+1:]...)
```

Comment:
doesn't this leave the Order attribute on all subsequent items off by one (and make us vulnerable to out-of-bounds panics if the last item in the queue is then deleted?

Reply:
yes, thanks for your review. this is a serious problem, the `order` in OrderItem will out of date after the pop or delete operation. It's hard to use `slice` to achieve delete in O(1), thus using `list` maybe a good way to impl.